contracts/instruments/RibbonCoveredCall.sol

```diff
@@ -264,15 +274,42 @@ contract RibbonCoveredCall is DSMath, OptionsVaultStorage {
     }
 
     /**
-     * @notice Sets the next option address and the timestamp at which the admin can call `rollToNextOption` to open a short for the option
-     * @param optionTerms is the terms of the option contract
+     * @notice Sets the next option the vault will be shorting, and closes the existing short. This allows all the users to withdraw if the next option is malicious.
      */
-    function setNextOption(
+    function commitAndClose(
         ProtocolAdapterTypes.OptionTerms calldata optionTerms
     ) external onlyManager nonReentrant {
+        _setNextOption(optionTerms);
+        _closeShort();
+    }
+
+    function closeShort() external nonReentrant {
+        _closeShort();
+    }
+
+    /**
+     * @notice Sets the next option address and the timestamp at which the admin can call `rollToNextOption` to open a short for the option.
+     * @param optionTerms is the terms of the option contract
+     */
+    function _setNextOption(
+        ProtocolAdapterTypes.OptionTerms calldata optionTerms
+    ) private {
+        if (isPut) {
+            require(
+                optionTerms.optionType == ProtocolAdapterTypes.OptionType.Put,
+                "!put"
+            );
+        } else {
+            require(
+                optionTerms.optionType == ProtocolAdapterTypes.OptionType.Call,
+                "!call"
+            );
+        }
+
         address option = adapter.getOptionsAddress(optionTerms);
         require(option != address(0), "!option");
         OtokenInterface otoken = OtokenInterface(option);
+        require(!otoken.isPut(), "!call");
```

Comment:
should change to require(otoken.isPut() == isPut, "!call/put");

otherwise wont let put options for wbtc/eth vault